api/web3server_marshal.go

```diff
@@ -229,12 +227,7 @@ func (obj *getTransactionResult) MarshalJSON() ([]byte, error) {
 	}
 	value, _ := intStrToHex(obj.ethTx.Value().String())
 	gasPrice, _ := intStrToHex(obj.ethTx.GasPrice().String())
-
-	// TODO: if transaction is support EIP-155, we need to add chainID to v
-	vVal := uint64(obj.signature[64])
-	if vVal < 27 {
-		vVal += 27
-	}
+	v, r, s := obj.ethTx.RawSignatureValues()
```

Comment:
This change may not be compatible when chainID is not 0, please add a test to verify

Reply:
it is compatible, and verified using the specific tx that you run into, added a test in latest commit